### ISSUE TITLE
a puzzle about selection and frequency trajectories

### ISSUE THREAD

**Issue by Armand1** [ben18785/Selection_simulations#12]
I think it would be nice if we could predict the frequency trajectories of individual species from their selection coefficients.  I can imagine how to do that using simulation.  So, I looked at the trajectories of the five most fit and five least fit species.  In general, they behave as they should:  fit species increase in frequency, unfit ones decrease.  But not all do. For example,  *Erythrina constaricensis*  is the fifth-most fit species, yet it declines quite strongly in frequency.  Any thoughts as to why this might be?

These trajectories are from the data that I gave you.  But you then bashed them into shape for your model.  It would be good to get the bashed data to make sure nothing funny happened there.

**Five fit species**
![fivefit](https://user-images.githubusercontent.com/8698079/60816556-df49db00-a191-11e9-9b2e-715855d818d3.jpeg)

**Five unfit species**
![fiveunfit](https://user-images.githubusercontent.com/8698079/60816555-deb14480-a191-11e9-8cca-78c72353f85c.jpeg)

**Comment by Armand1** [comment]
Now that I have the names of the species I can check whether they reflect linear regression coefficients ("estimate" on the x axis below). It doesn't look good. 

![Rplot02](https://user-images.githubusercontent.com/8698079/61466171-5a20ac00-a971-11e9-82c3-ff73a302ece1.jpeg)

For my full analysis see this:

[checking_estimates_of_selection_on_species.pdf](https://github.com/ben18785/Selection_simulations/files/3406976/checking_estimates_of_selection_on_species.pdf)

I can't help but think that the names have somehow gotten scrambled in the selection coefficient output

**Comment by ben18785** [comment]
Ok, yes, that doesn't look right. Let me check!

**Comment by Armand1** [comment]
An additional puzzle.  

I estimated the regression coefficients on your bashed data.  They disagree with your selection coefficients.  That's a problem. 

However, I also decided to check the bashed data against the original data that I gave you.  Here are two plots. Each point is a count for a species in a given year.  They should be identical -- have a slope = 1 -- except for those species which have been given subscripts (XXX_1) which would have been excluded by the merge.   But they're not. 

![Rplot06](https://user-images.githubusercontent.com/8698079/61471071-05cdfa00-a97a-11e9-9d31-c4990972087d.jpeg)

here's one particularly common species. 

![Rplot07](https://user-images.githubusercontent.com/8698079/61471070-05cdfa00-a97a-11e9-8d8e-4e57ebe06102.jpeg)
I just don't get it. For this species (Hesteria concinna) your bashed data has 30-40,000 individuals per year. The data I sent you has a few hundred. 

Here's confirmation that the data have been scrambled.  Your "Heisteria concinna" has the data for my "Hyabanthus prunifolius"

Data_bashed_by_Ben
Heisteria concinna                38697
Heisteria concinna                41510
Heisteria concinna                41016
Heisteria concinna                36886
Heisteria concinna                32723
Heisteria concinna                28343
Heisteria concinna                26869

Data given to Ben
Hybanthus prunifolius_1982               38697
Hybanthus prunifolius_1985               41088 
Hybanthus prunifolius_1990               40439 
Hybanthus prunifolius_1995               36058 
 Hybanthus prunifolius_2000               31925 
 Hybanthus prunifolius_2010               27845 
Hybanthus prunifolius_2015               26869

**Note**.  The name scrambling has happened **twice**
First, from my "BCI data forben reproductives only" --> your (bashed) "reprodutives names_stan_rds.)

Second, from your bashed data --> the selection coefficients.  The latter must be true since I estimated the linear regressions from your bashed data and they don't agree either. 

So, implausible though it may seem, there have been two name errors.  Unless I've fucked all this up!

**Comment by ben18785** [comment]
Yes, this is all weird, sorry. I am looking through it now and have found
the same thing.

**Comment by ben18785** [comment]
Right; sorry. It was an issue with the newly introduced species messing up
the order. I have checked the counts on the new data frame versus the old
and all (I think) seems ok. Can you rerun your analyses with the new data?
Note there are new selection estimates too as the names were messed up in
that file as well.

Ben

**Comment by Armand1** [comment]
ok --point me to the new data?

**Comment by ben18785** [comment]
Oh, thought it was attached. See your email.

**Comment by Armand1** [comment]
The bashed data are fixed relative to the data I gave you. So that's good!

The bad news, I think, is that the selection names are still not fixed relative to the bashed data. 
This is a version of the same plot as above, just with the axes flipped (it seems more intuitive). 

We'd expect that s_freqinde-50 increases the estimate should become more positive.  But it doesn't. 
I have been racking my brains wondering if we are misunderstanding this in some way, but I don't see it.  Can you look at the selection names again?
![Rplot04](https://user-images.githubusercontent.com/8698079/61482805-6584cf00-a993-11e9-8708-b33dbe4e011e.jpeg)

While you're about it could you make the selection coefficients relative to max(s_freqinde-50)? rather than max(s_freqinde) or max(s_freqinde-97.5) which it is now? I want the main estimate of the best species to be zero.

**Comment by ben18785** [comment]
I'm not sure what you're plotting. What is "estimate"? There doesn't seem
to be much variation in this quantity...

**Comment by Armand1** [comment]
See pdf above - it’s the slope of a regression of frequency v census year.
One for every species.

**Comment by ben18785** [comment]
This is a graph of my estimated beta coefficients vs the % change in frequency (from 1982-2010) for each variant. To me, this looks like the estimation is working.

[freq_inde_beta_vs_change.pdf](https://github.com/ben18785/Selection_simulations/files/3411439/freq_inde_beta_vs_change.pdf)

Similarly, for estimated selection coefficients from the independent model.

[freq_inde_s_vs_change.pdf](https://github.com/ben18785/Selection_simulations/files/3411445/freq_inde_s_vs_change.pdf)

@Armand1 What do you think?

**Comment by Armand1** [comment]
Well, scrambled data could hardly produce that result.  Then there’s something funny about the relationship between selection coefficients and regression coefficients. I’ll probe a bit more.
